extensions/sql/policy/store/src/test/java/org/eclipse/dataspaceconnector/slq/policy/store/SqlPolicyStoreTest.java

```diff
@@ -0,0 +1,212 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class SqlPolicyStoreTest {
+
+    private static final String DATASOURCE_NAME = "policy";
+
+    private SqlPolicyStore sqlPolicyStore;
+    private ConnectionPool connectionPool;
+
+    @BeforeEach
+    void setUp() throws SQLException {
+        var monitor = new ConsoleMonitor();
+        var txManager = new LocalTransactionContext(monitor);
+        DataSourceRegistry dataSourceRegistry;
+        dataSourceRegistry = new LocalDataSourceRegistry(txManager);
+        var transactionContext = (TransactionContext) txManager;
+        var jdbcDataSource = new JdbcDataSource();
+        jdbcDataSource.setURL("jdbc:h2:mem:");
+
+        var connection = jdbcDataSource.getConnection();
+        var dataSource = new ConnectionFactoryDataSource(() -> connection);
+        connectionPool = new CommonsConnectionPool(dataSource, CommonsConnectionPoolConfig.Builder.newInstance().build());
+        var poolDataSource = new ConnectionPoolDataSource(connectionPool);
+        dataSourceRegistry.register(DATASOURCE_NAME, poolDataSource);
+        txManager.registerResource(new DataSourceResource(poolDataSource));
+        sqlPolicyStore = new SqlPolicyStore(dataSourceRegistry, DATASOURCE_NAME, transactionContext, new TypeManager(), new PostgressStatements());
+
+        try (var inputStream = getClass().getClassLoader().getResourceAsStream("schema.sql")) {
+            var schema = new String(Objects.requireNonNull(inputStream).readAllBytes(), StandardCharsets.UTF_8);
+            transactionContext.execute(() -> SqlQueryExecutor.executeQuery(connection, schema));
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        connectionPool.close();
+    }
+
+    @Test
+    @DisplayName("Save a single policy that not exists")
+    void save_notExisting() {
+        var policy = getDummyPolicy(getRandomId());
+
+        sqlPolicyStore.save(policy);
+
+        var policyFromDb = sqlPolicyStore.findById(policy.getUid());
+        assertThat(policyFromDb).isNotNull();
+        assertThat(policy.getUid()).isEqualTo(policyFromDb.getUid());
+    }
+
+    @Test
+    @DisplayName("Save a single a single policy that already exists")
+    void save_alreadyExists() {
+        var id = getRandomId();
+        var policy = getDummyPolicy(id);
+
+        sqlPolicyStore.save(policy);
+
+        assertThatThrownBy(() -> sqlPolicyStore.save(policy)).isInstanceOf(EdcPersistenceException.class);
+    }
+
+    @Test
+    @DisplayName("Find policy by ID that exists")
+    void findById_whenPresent() {
+        Policy policy =  getDummyPolicy(getRandomId());
+        sqlPolicyStore.save(policy);
+
+        var policyFromDb = sqlPolicyStore.findById(policy.getUid());
+
+        assertThat(policy.getUid()).isEqualTo(policyFromDb.getUid());
```

Comment:
this performs a stronger assertion:
```suggestion
assertThat(policyFromDb).usingRecursiveComparison().isEqualTo(policy);
```

also you got `actual` and `expected` mixed up

Reply:
Will do so.